src/Interpreters/ExpressionActions.cpp

```diff
@@ -243,7 +245,18 @@ void ExpressionAction::prepare(Block & sample_block, const Settings & settings,
 
         case ARRAY_JOIN:
         {
-            array_join->prepare(sample_block);
+            ColumnWithTypeAndName current = sample_block.getByName(source_name);
```

Comment:
Well, I think one-line functions are more readable.
However, it would be reasonable rewrite the whole function then. I would rather do it next time.

Also, I don't think this logic is complex, especially comparing with previous many-column implementation.